tests/cephfs/snapshot_clone/cg_snap_test.py

```diff
@@ -0,0 +1,879 @@
+
+                if test_fail == 1:
+                    i = repeat_cnt
+                else:
+                    i += 1
+                    time.sleep(10)
+            else:
+                i = repeat_cnt
+
+        log.info(f"cg_test_io_status : {cg_test_io_status.value}")
+
+        if p.is_alive():
+            proc_stop = 0
+            log.info("CG IO is running after quiesce lifecycle")
+            end_time = datetime.datetime.now() + datetime.timedelta(seconds=io_run_time)
+            while (datetime.datetime.now() < end_time) and (proc_stop == 0):
+                if p.is_alive():
+                    time.sleep(10)
+                else:
+                    proc_stop = 1
+            if proc_stop == 1:
+                log.info("CG IO completed")
+            elif proc_stop == 0:
+                raise Exception("CG IO has NOT completed")
+        else:
+            log.info(
+                f"WARN:CG IO test completed early during quiesce lifecycle with await on qs_set {qs_id_val}"
+            )
+
+        mnt_pt_list = []
+        log.info(f"Perform cleanup for {qs_set}")
+        for qs_member in qs_member_dict1:
+            mnt_pt_list.append(qs_member_dict1[qs_member]["mount_point"])
+        log.info("Remove CG IO files and unmount")
+        cg_snap_util.cleanup_cg_io(client1, mnt_pt_list)
+        mnt_pt_list.clear()
+        for qs_member in qs_member_dict2:
+            mnt_pt_list.append(qs_member_dict2[qs_member]["mount_point"])
+        log.info("Unmount")
+        cg_snap_util.cleanup_cg_io(client2, mnt_pt_list, del_data=0)
+        snap_name = f"cg_snap_{rand_str}"
+        log.info("Remove CG snapshots")
+        for qs_member in qs_member_dict1:
+            snap_list = snap_qs_dict[qs_member]
+            if qs_member_dict1[qs_member].get("group_name"):
+                group_name = qs_member_dict1[qs_member]["group_name"]
+                for snap_name in snap_list:
+                    fs_util.remove_snapshot(
+                        client,
+                        fs_name,
+                        qs_member,
+                        snap_name,
+                        validate=True,
+                        group_name=group_name,
+                    )
+            else:
+                for snap_name in snap_list:
+                    fs_util.remove_snapshot(
+                        client, fs_name, qs_member, snap_name, validate=True
+                    )
+
+        if cg_test_io_status.value == 1:
+            log.error(
+                f"CG IO test exits with failure during quiesce lifecycle with await on qs_set-{qs_id_val}"
+            )
+            test_fail = 1
+
+    if test_fail == 1:
+        log.error(
+            "FAIL: Workflow 2 - Verify quiesce release with if-version,with exclude,include prior to release"
+        )
+        return 1
+    return 0
+
+
+####################
+# HELPER ROUTINES
+####################
+def umount_all(mnt_paths, umount_params):
```

Comment:
This function is not used anywhere suma. Does this need to be used in finally block

Reply:
Thanks Amar, i removed this code.